skfda/representation/grid.py

```diff
@@ -741,11 +741,14 @@ def _get_op_matrix(
             self._check_same_dimensions(other)
             return other.data_matrix
 
+        elif isinstance(other, Callable):
+            return np.array([[other(x) for x in _gp] for _gp in self.grid_points]).T
```

Comment:
A couple of things:
- I think we should limit support only to vectorized functions. Otherwise we have to call repeteadly the function for every point, which is too slow.
- I think this code won't work for functions of several variables (e.g. surfaces). For supporting that case we should probably create a matrix with every coordinate were functions are going to be evaluated, call the function with that matrix and then reshape the result.
- Another option would be to check first if the function supports vectorized input, and fall back to the slow point by point evaluation otherwise. This would be (much) less performant, but probably useful for newcomers and small datasets.
- `FData` are callables too. What should happen if we do ops between `FData` with different domains? And between two `FDataGrid` with the same domain and different `grid_points`? (we could lose commutativity in this case!) And between a `FDataGrid` and a `FDataBasis` with the same domain?

Reply:
1. My understanding is that it doesn't make a real difference in terms of performance except for very small arrays, see [1].
2/3. This is true, it's now fixed.
4. This is also true, and a good point, which is why the testing for `Callable` is after testing for `FDataGrid`. That way all your concerns are tackled by the `FDataGrid` handler.

[1] https://stackoverflow.com/questions/35215161/most-efficient-way-to-map-function-over-numpy-array

Reply:
Sorry for the delay.

> 1. My understanding is that it doesn't make a real difference in terms of performance except for very small arrays, see [1].

I didn't mean to use `np.vectorize`, but to assume that the `other` function accepts an array as input and return the array as output. This way, it would be very efficient for ufuncs like `np.sqrt` or `np.sin`, or functions created composing these, as they perform the iteration in C code, probably using SIMD instructions for further performance.

>        2/3. This is true, it's now fixed.

I would first try passing the  array of coordinates, and if that fails pass the coordinates as different parameters for the efficiency reasons explained above.

>     2. This is also true, and a good point, which is why the testing for `Callable` is after testing for `FDataGrid`. That way all your concerns are tackled by the `FDataGrid` handler.

Ok, so for now `FDataGrid` is special, and two `FDataGrid`s with different domains cannot be operated together. If a user wants that they can always use `lambda x: my_fdatagrid(x)` instead. We can also change this behaviour in the future if we found a reasonable way to support it.